AtReconstruction/AtFilter/AtSCACorrect.h

```diff
@@ -0,0 +1,58 @@
+#ifndef ATSCACORRECT_H
+#define ATSCACORRECT_H
+
+/**
+ * A filter to perform baseline subtraction using the average baseline and average phase effect stored as
+ * AtRawEvent objects in a separate file or files.
+ *
+ */
+
+#include "AtFilter.h"
+
+#include "TFile.h"
+#include "TString.h"
+
+#include <iostream>
+
+class AtMap;
+
+using AtMapPtr = std::shared_ptr<AtMap>;
+
+class AtSCACorrect : public AtFilter {
+private:
+   AtMapPtr fMap;
+
+   AtRawEvent *fBaseline;
+   AtRawEvent *fPhase;
```

Comment:
This memory will be leaked. It needs to be cleaned up on object destruction. So either should be objects, not pointers to object, or a std::unique_ptr. I'm going to assume we go the object route and give suggestions for the constructor below.